llvm/lib/CodeGen/TargetPassConfig.cpp

```diff
@@ -1182,6 +1186,11 @@ bool TargetPassConfig::addISelPasses() {
     }
     addPass(createYkSplitBlocksAfterCallsPass());
   }
+  // `YkModuleClone` needs to run before `YkBasicBlockTracerPass` and
+  // `YkInsertStackMaps` because they both skip cloned functions.
+  if (YkModuleClone) {
+    addPass(createYkModuleClonePass());
```

Comment:
This is not a bad spot for the cloning pass. But I think we might want to run it even earlier before the block-splitting and no calls in entry blocks pass (and maybe also before the blockdisambiguate pass). Though that means we need to manually add the control point and a single stackmap for that control point into the cloned module. We need those in order to jump between the two modules.

Reply:
Got it.
Made changes here :point_right: https://github.com/ykjit/ykllvm/pull/202/commits/c1225dfcce0f3be65b3c8c0840123eaff4f15adf. 
I think it actually works by reusing the same `YkStackmaps` pass with different configurations for `stackmapIDStart` - 1 by default, 2 in case of `YkModuleClone` is active.

Reply:
Excellent. Does this still insert stackmaps into the cloned module? The cloned module only needs a single stackmap at the control point. Though, while important, it's kind of an optimisation, so you may choose to do this in another PR.

Reply:
yes, the the patchpoint emits stackmaps.
For the cloned module there's a patchpoint in the original main function
```
18:                                               ; preds = %15
  call void @__yk_trace_basicblock(i32 2, i32 8), !dbg !57
  %19 = load ptr, ptr %4, align 8, !dbg !57
  call void @llvm.dbg.value(metadata ptr %19, metadata !32, metadata !DIExpression()), !dbg !27
  call void (i64, i32, ptr, i32, ...) @llvm.experimental.patchpoint.void(i64 0, i32 13, ptr @__ykrt_control_point, i32 3, ptr %19, ptr %3, i64 0, ptr %3, ptr %4, ptr %5, ptr %19), !dbg !59
  br label %20, !dbg !60
```
and cloned main function:
```shell
18:                                               ; preds = %15
  %19 = load ptr, ptr %4, align 8, !dbg !97
  call void @llvm.dbg.value(metadata ptr %19, metadata !80, metadata !DIExpression()), !dbg !76
  call void (i64, i32, ptr, i32, ...) @llvm.experimental.patchpoint.void(i64 1, i32 13, ptr @__ykrt_control_point, i32 3, ptr %19, ptr %3, i64 1, ptr %3, ptr %4, ptr %5, ptr %19), !dbg !99
  br label %20, !dbg !100
```